ast/env.go

```diff
@@ -22,6 +23,23 @@ func NewTypeEnv() *TypeEnv {
 	}
 }
 
+// WithSchemas sets the user-provided schemas
+func (env *TypeEnv) WithSchemas(schemas *SchemaSet) *TypeEnv {
+	env.schemaSet = schemas
+	return env
+}
+
+// GetExistingPrefix returns the shortest prefix of ref that exists in env
```

Comment:
nit: I'd probably rename this `GetPrefix` since the "existing" is a bit redundant.

Reply:
done